Fabric/src/main/java/me/ferdz/placeableitems/init/PlaceableItemsBlockRegistry.java

```diff
@@ -218,8 +228,16 @@ public static void onBlocksRegistry() {
                         .build()
                         .setShape(VoxelShapesUtil.create(8, 12, 8))
                         .register("dragon_breath_block", Items.DRAGON_BREATH),
-                // TODO: Breaks with a 1/8 chance of spawning a chicken when right-clicked
                 EGG = new PlaceableItemsBlockBuilder()
+                        .addComponent(new FragileBlockComponent().withEntity((state, world, pos, player, hand, hit) -> {
+                            if (world.random.nextInt(8) == 0) {
+                                ChickenEntity chicken = EntityType.CHICKEN.create(world);
+                                chicken.setBreedingAge(-24000);
+                                return chicken;
+                            }
+
+                            return null;
+                        }))
```

Comment:
Interaction code shouldn't be defined in the declaration class. It would be better to create a dedicated `BlockComponent` for this. Something like `EggBlockComponent`

Reply:
The issue with that is it ends up being similar to the BoneBlockComponent in that it's a component specific to one block. Is that still something that should be encouraged?

Reply:
Not really something we want to encourage, but there's no other case in which spawning an entity on right click I can think of. Actually I'm not sure if this behavior should be combined with "drop and item stack on right click", since they're both unrelated

Reply:
Might I propose 3 different components?
- `BlockComponentFragile`: Breaks when right clicked
- `BlockComponentItemStackSource`: Drops an ItemStack when right clicked
- `BlockComponentEntitySource`: Summons an entity when right clicked
I'm not too strung up on the names either, open to suggestions. This way, egg can have both a fragile and entity source component whereas the bone can have a fragile and item stack source component.

Reply:
Didn't think of that, sounds great :)